core/templates/domain/skill/skill-backend-api.service.ts

```diff
@@ -16,124 +16,142 @@
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService
+  ) {}
+
+  private _fetchSkill(
+      skillId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    const skillDataUrl = this.urlInterpolationService.interpolateUrl(
+      SkillDomainConstants.EDITABLE_SKILL_DATA_URL_TEMPLATE, {
+        skill_id: skillId
+      }
+    );
+    this.http.get(skillDataUrl).toPromise().then(
+      (data: EditableSkillResponseConfig) => successCallback({
+        skill: data.skill,
+        groupedSkillSummaries: data.grouped_skill_summaries
+      }),
+      error => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(error.error);
         }
-      });
-    };
-
-    var _fetchMultiSkills = function(skillIds, successCallback, errorCallback) {
-      var skillDataUrl = UrlInterpolationService.interpolateUrl(
-        SKILL_DATA_URL_TEMPLATE, {
-          comma_separated_skill_ids: skillIds.join(',')
-        });
+      }
+    );
+  }
 
-      $http.get(skillDataUrl).then(function(response) {
-        var skills = angular.copy(response.data.skills);
-        if (successCallback) {
-          successCallback(skills);
-        }
-      }, function(errorResponse) {
+  private _fetchMultiSkills(
+      skillIds: Array<string>,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    const skillDataUrl = this.urlInterpolationService.interpolateUrl(
+      SkillDomainConstants.SKILL_DATA_URL_TEMPLATE, {
+        comma_separated_skill_ids: skillIds.join(',')
+      }
+    );
+    this.http.get(skillDataUrl).toPromise().then(
+      (data: EditableSkillResponseConfig) => successCallback(data.skills),
+      error => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(error.error);
         }
-      });
-    };
-
-    var _updateSkill = function(
-        skillId, skillVersion, commitMessage, changeList,
-        successCallback, errorCallback) {
-      var editableSkillDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_SKILL_DATA_URL_TEMPLATE, {
-          skill_id: skillId
-        });
+      }
+    );
+  }
+  private _updateSkill(
+      skillId: string, skillVersion: number, commitMessage: string,
+      changeList: Array<ChangeList>,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    const editableSkillDataUrl = this.urlInterpolationService.interpolateUrl(
+      SkillDomainConstants.EDITABLE_SKILL_DATA_URL_TEMPLATE, {
+        skill_id: skillId
+      }
+    );
 
-      var putData = {
-        version: skillVersion,
-        commit_message: commitMessage,
-        change_dicts: changeList
-      };
+    const putData: UpdateSkillPayload = {
```

Comment:
`UpdateSkillPayload` 

We follow the  `I<name` for interface name, isn't it? (Currently, it looks like `UpdateSkillPayload` is a service that will update the skill payload.)

Also, only start the function name with verb. (Maybe `ISkillUpdatePayoad` (not sure))